File: swift/Sources/MasterMind/CheckResult.swift
//
//  CheckResult.swift
//  MasterMind
//
//  Created by oglimmer on 11/12/21.
//

import Foundation

class CheckResult {
    var hit: Int = 0
    var colorOnlyHit: Int = 0
    
    func getHit() -> Int {
        return hit;
    }
    func getColorOnlyHit() -> Int {
        return colorOnlyHit;
    }
    func incHit() {
        hit+=1
    }
    func incColorOnlyHit() {
        colorOnlyHit+=1
    }
    func hasWon() -> Bool {
        return getHit() == 4
    }
}

File: js_oo/CheckResult.js
export class CheckResult {

    hit = 0
    colorOnlyHit = 0

    getHit() {
        return this.hit;
    }
    
    getColorOnlyHit() {
        return this.colorOnlyHit;
    }
    
    incHit() {
        this.hit+=1
    }
    incColorOnlyHit() {
        this.colorOnlyHit+=1
    }

    hasWon() {
        return this.getHit() == 4
    }

}